state/protocol/protocol_state/kvstore/models.go

```diff
@@ -401,9 +401,43 @@ func NewDefaultKVStore(finalizationSafetyThreshold, epochExtensionViewCount uint
 	if err != nil {
 		return nil, fmt.Errorf("could not construct v0 kvstore: %w", err)
 	}
-	return &Modelv1{
-		Modelv0: *modelv0,
-	}, nil
+	return &Modelv2{Modelv1: Modelv1{Modelv0: *modelv0}}, nil
+}
+
+// NewKVStore constructs a key-value store for a particular Protocol State version for bootstrapping.
+func NewKVStore(
+	version uint64,
+	finalizationSafetyThreshold, epochExtensionViewCount uint64,
+	epochStateID flow.Identifier,
+) (protocol_state.KVStoreAPI, error) {
+	modelv0, err := newKVStoreV0(finalizationSafetyThreshold, epochExtensionViewCount, epochStateID)
+	if err != nil {
+		return nil, fmt.Errorf("could not construct v0 kvstore: %w", err)
+	}
+
+	switch version {
+	case 0:
+		return modelv0, nil
+	case 1:
+		return &Modelv1{Modelv0: *modelv0}, nil
+	case 2:
+		return &Modelv2{Modelv1: Modelv1{Modelv0: *modelv0}}, nil
+	case 3:
```

Comment:
version 3 has not been implemented yet. We shouldn't allow creating it by any means, at least so far.